azurecontainerapps.ts

```diff
@@ -150,8 +154,12 @@ export class azurecontainerapps {
         // Get the YAML configuration file, if provided
         this.yamlConfigPath = this.toolHelper.getInput('yamlConfigPath', false) as string;
 
+        // Get the name of the image to build, if provided
+        this.imageToBuild = this.toolHelper.getInput('imageToBuild', false) as string;
+
         // Ensure that acrName or registryUrl is also provided if appSourcePath is provided
-        if (!this.util.isNullOrEmpty(this.appSourcePath) && this.util.isNullOrEmpty(this.acrName) && this.util.isNullOrEmpty(this.registryUrl)) {
+        // and if imageToBuild is either empty or doesn't start with 'default/' (i.e. it's not source to cloud scenario)
+        if (!this.util.isNullOrEmpty(this.appSourcePath) && this.util.isNullOrEmpty(this.acrName) && this.util.isNullOrEmpty(this.registryUrl) && (this.util.isNullOrEmpty(this.imageToBuild) || !this.imageToBuild.startsWith('default/'))) {
```

Comment:
How would users not going through the Portal UX leverage the source-to-cloud functionality in the Azure CLI? It seems like they would need to know to provide a random `default/*` value for `imageToBuild` or else this validation will fail their workflow.

Reply:
Users should be able to trigger source to cloud from the CLI by executing `az containerapp up --source ...` by meeting these conditions. 
1. No Docker running/installed.
2. No registry available/provided.

They don't need to provide a random `default/*` image. This image is needed only for triggering cloud build from source control.

Reply:
Let me rephrase the question: if the user wants to use the cloud build service with an internal registry, they are providing their application source and _are not_ providing an ACR or external registry, so their workflow will fail due to this validation in all cases _unless_ they provide a value for `imageToBuild` that is `default/*`, right? That's how this conditional seems to operate based on my interpretation.

Reply:
Thank you for clarifying. Sorry I misunderstood your question.

I hope I'm not missing anything else, but the source control API should supply the `imageToBuild` arg starting with `default/* in the template that it sets up while calling the action`?

That's how this action will be called? So here is how the flow looks like from my understanding - 

Portal
Customer creates container app/sets source control by choosing `Push to internal registry` -> Source Control API performs validation (skips mandatory check for registry because image matches `default/*` -> Calls action from the template and supplies the `imageToBuild` arg starting with `default/*` so this isn't something that is set by the customer.

Reply:
> I hope I'm not missing anything else, but the source control API should supply the `imageToBuild` arg starting with `default/*` in the template that it sets up while calling the action?

This is correct, but 

> If customers were to use the internal registry, they would directly just run the CLI commands?

I don't think this is correct -- there are a lot of customers who use the action independent of the source control API, so we need to make sure that if any of those customers want to use the internal registry functionality that they can do so without needing to go through the source control API to provide them a `default/*` value or without telling them that they need to use the CLI themselves. Does that makes sense?

Reply:
Synced offline. We don't need to validate against `imageToBuild` in the GitHub action because if `appSourcePath` is provided and both `acrName` and `registryUrl` are Null, we default to the internal registry flow.